modules/test/playwright/tests/frontend-data-set-views-web/pages/ActionsPage.ts

```diff
@@ -51,7 +51,11 @@ export class ActionsPage {
 		await this.viewsPage.goto();
 		await this.viewsPage.gotoSampleDataSetView();
 
-		this.page.getByRole('button', {name: 'Actions'}).first().click();
+		await this.page
+			.locator('li')
```

Comment:
Continuing on https://github.com/liferay-frontend/liferay-portal/pull/3978/files#r1481514084:

This looks very fragile. Could this be fixed with an explicit `expect`? It would confirm that the correct page is loaded. `expect` on something unique on the page, not necessary that list item. The check can be in `gotoSampleDataSetView`, or here.

This is probably a very common issue, but typically not seen, as after `goto*` is some specific locator.

Reply:
I'd say the  `goto()` in the corresponding poms should make sure to take navigation to the expected place before finishing. This may be an issue in SPAs where gotos are not really a URL-based navigation but rather do some interaction in the UI that requires app to trigger a (xhr) fetch+render cycle.

Not checked this but I highly suspect we'll need to apply some technique here (like the ones proposed in https://blog.apify.com/scraping-single-page-applications-with-playwright/) to ensure `goto()` are reliable. Current playwright api led me to this possibility → https://playwright.dev/docs/api/class-page#page-wait-for-function